plugins/nanospell/plugin.js

```diff
@@ -18,769 +18,770 @@
+				}
+			}
+
+			function ignoreWord(target, word, all) {
+				var i;
+				if (all) {
+					ignorecache[word.toLowerCase()] = true;
+					for (i in suggestionscache) {
+						if (i.toLowerCase() == word.toLowerCase()) {
+							delete suggestionscache[i];
+						}
+					}
+					var allInstances = editor.document.find('span.nanospell-typo');
+					for (i = 0; i < allInstances.count(); i++) {
+						var item = allInstances.getItem(i);
+						var text = item.getText();
+						if (text == word) {
+							item.remove(true);
+						}
+					}
+				} else {
+					target.remove(true);
+				}
+			}
+
+			function overrideCheckDirty() {
+
+				var editorCheckDirty = CKEDITOR.editor.prototype;
+
+				editorCheckDirty.checkDirty = CKEDITOR.tools.override(editorCheckDirty.checkDirty, function (org) {
+
+					return function () {
+						var retval = (this.status == 'ready');
+
+						if (retval) {
+							var currentData = clearAllSpellCheckingSpansFromString(this.getSnapshot()),
+								prevData = clearAllSpellCheckingSpansFromString(this._.previousValue);
+
+							retval = (retval && (prevData !== currentData))
+						}
+
+						return retval;
+					};
+				});
+
+				editorCheckDirty.resetDirty = CKEDITOR.tools.override(editorCheckDirty.resetDirty, function (org) {
+					return function () {
+						this._.previousValue = clearAllSpellCheckingSpansFromString(this.getSnapshot());
+					};
+				});
+			}
+
+
+		},
+		addRule: function (editor) {
+			var dataProcessor = editor.dataProcessor,
+				htmlFilter = dataProcessor && dataProcessor.htmlFilter,
+				pathFilters = editor._.elementsPath && editor._.elementsPath.filters,
+				dataFilter = dataProcessor && dataProcessor.dataFilter,
+				removeFormatFilter = editor.addRemoveFormatFilter,
+				pathFilter = function (element) {
+					if (element.hasClass('nanospell-typo')) {
+						return false;
+					}
+				},
+				removeFormatFilterTemplate = function (element) {
+					var result = true;
+
+					if (element.hasClass('nanospell-typo')) {
+						result = false;
+					}
+
+					return result;
+				};
+
+			if (pathFilters) {
+				pathFilters.push(pathFilter);
+			}
+
+			if (dataFilter) {
+				var dataFilterRules = {
+					elements: {
+						span: function (element) {
+
+							var scaytState = element.hasClass('nanospell-typo');
+
+							if (scaytState) {
+								delete element.name;
+							}
+
+							return element;
+						}
+					}
+				};
+
+				dataFilter.addRules(dataFilterRules);
+			}
+
+			if (htmlFilter) {
+				var htmlFilterRules = {
+					elements: {
+						span: function (element) {
+
+							var scaytState = element.hasClass('nanospell-typo');
+
+							if (scaytState) {
+								delete element.name;
+							}
+
+							return element;
+						}
+					}
+				};
+
+				htmlFilter.addRules(htmlFilterRules);
+			}
+
+			if (removeFormatFilter) {
+				removeFormatFilter.call(editor, removeFormatFilterTemplate);
+			}
+		},
+		hasPersonal: function (word) {
```

Comment:
everything below was extracted mostly for testability.  some of them probably don't need to be public.